src/packet/BinaryWriter.js

```diff
@@ -1,137 +1,62 @@
-﻿'use strict';
+'use strict';
 /*
  * Simple BinaryWriter is a minimal tool to write binary stream with unpredictable size.
  * Useful for binary serialization.
  *
- * Copyright (c) 2016 Barbosik
- * 
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * 
- *     http://www.apache.org/licenses/LICENSE-2.0
- * 
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- * 
- * 
+ * Copyright (c) 2016 Barbosik https://github.com/Barbosik
+ * License: Apache License, Version 2.0
  */
 
-function BinaryWriter(size) {
-    if (!size || size <= 0) {
-        size = Buffer.poolSize / 2;
-    }
-    this._buffer = new Buffer(size);
+function BinaryWriter() {
+    this._buffer = parseInt(process.version[1]) < 6 ? new Buffer(1e5) : Buffer.allocUnsafe(1e5);

```

Comment:
1. You can't crash the server with this exploit. The server can only crash itself with this problem. Think good about what you just said.
2. 100 000 bytes is a good limit.
3. 100 000 bytes can send up to 5 553 pellet cells through via protocol 6. Compare that with the average visible cell count for a 100 000 mass player, 940.
4. And if it ever overpopulates we can use allocation checks again.
4. Barbosik's allocation checks were expensive.
5. A new BinaryWriter instance is created for every packet.
6. Pre-calculating byte count requires listing through every string's characters twice.
7. This thing is fast enough for our needs.